poetry/utils/helpers.py

```diff
@@ -27,19 +27,18 @@ def normalize_version(version):  # type: (str) -> str
     return str(Version(version))
 
 
+def del_ro(action, name, exc):
```

Comment:
Having a check that making the file writable will fix the error raised by rmtree (or that the error was raised because of being nonwriteable) wouldn't hurt. In case different error ocurred, making a file writable might be unwanted? But that's just me guessing.

Reply:
Considering this is being applied to a temp directory, specifically one being used for unpacking sdist's, I doubt making something writeable before deletion would hurt anything.

Reply:
Thinking about it, my comment really comes from worrying that the code isn't self-documenting in this case and might benefit from a comment explaining why there's the `onerror, what is it supposed to handle... But that's just my personal feeling.

Reply:
Personally, I feel that the "what" is clear (delete read-only files), the problem is the "why" which is that some sdist's contain read-only files.

The question is whether that kind of "why" lives better in a comment or in a commit message.  Personally, this feels more like the type of historical information that is best left in a commit message but I can also see the case for why it could be a comment.

So I'm fine changing this if its what is needed to get this PR accepted.